nemo_rl/models/policy/workers/megatron_policy_worker.py

```
@@ -721,139 +689,385 @@ def generate(
             GPTInferenceWrapper,
         )
-        from megatron.core.inference.sampling_params import SamplingParams
-
-        mcore_generation_config = cast(
-            MegatronGenerationConfig, self.cfg["generation"]["mcore_generation_config"]
+        from megatron.core.inference.text_generation_controllers.text_generation_controller import (
+            TextGenerationController,
         )
-        buffer_size_gb = mcore_generation_config["buffer_size_gb"]
 
+        model_cfg = self.megatron_cfg.model
```

Ruff (0.15.1)

[error] Local variable `model_cfg` is assigned to but never used

Remove assignment to unused variable `model_cfg`

(F841)